### PR TITLE
configuration-sample-script.sh 

### PR DESCRIPTION
**Pull request by Tobias-Kohl**
Added the script to compare 'ods-configuration-sample' and 'ods-configuration', also needs the rsync module in cygwin, which is now added to the cygwin installation guide in getting started . 

Also added a reminder in GettingStarted: 'Prepare CD project for Jenkins' to encode CD_USER_ID and CD_USER_PWD in ods-configuration.env with Base64.

### CODE REVIEW

**Comment by michaelsauter** [comment]
1. Do we really need `rsync` for this? Would appreciate if we can avoid it. I think we can simple remove all files ending in `.sample` and copy the new ones over ...
2. This should live in `ods-core` I think

**Comment by Tobias-Kohl** [comment]
1. All files with .sample already get cleaned and the ".sample"-Suffix removed. The problem is, if we dont use rsync we will override even those files, which did not get changed and just override all old files. As I understood the Use Case it was important to only change the new files, so that the old files with configurations stay intact. Else the user would need to reconfigure all .sample files again after every update.

2. Gonna change the location of the script to ods core and start new Pull Request there.

**Comment by michaelsauter** [comment]
@Tobias-Kohl I think the user should never touch the `.sample` files - they are samples after all. Maybe I did not get the script correctly, but I would:

* first time invocation, copy `.sample` over to new repo, then copy each file by removing the `.sample` suffix
* further invocations: copy `.sample` files, overwriting the existing `.sample` files - and do not touch the `.env` files created by the user ...

**Comment by Tobias-Kohl** [comment]
Ok, then that makes sense. Currently we expect the user to create ods-configuration and ods-configuration-sample. 
I understood the script should only copy the `.env` files after removing `.sample ` into ods-configuration. So that the current repository state can always be pulled into ods-configuration-sample and then the script only overrides the changed files into ods-configuration as `.env` files. But if we only copy over the `.sample` files and dont touch the `.env` files the current configuration stay intact. 
The User would then still need to touch the `.sample` files in his ods-configuration folder to update the `.env` files by hand.

**Comment by michaelsauter** [comment]
Yup, that was my understanding. `ods-configuration-sample` is a read-only repo, and `ods-configuration` is fully under the control of the user - but for convenience we can copy the `.sample` files there.

@tjaeschke Was that your understanding too?

**Comment by tjaeschke** [comment]
@michaelsauter Yes the configuration sample repo should be read only.  At the moment the script snippet provided in the getting started renames the copied values in ods-configuration, but it makes more sense to keep the samples as well in the ods-configuration.

**Comment by Tobias-Kohl** [comment]
Ok then i will change the script to the intended behaviour and add it into `ods-core`. Already removed the script from this pull request.

File: doc/getting-started.md [hunk 1]
```diff
@@ -595,6 +595,8 @@ find ods-configuration -name '*.sample' -type f | while read NAME ; do mv "${NAM
 
 Now you will have to check the `.env` configuration files in `ods-configuration`. Change all values with the suffix `_base64` to a Base64 encoded value.  
 
+If you want later changes to the repositories, update `ods-configuration-sample` and copy the configuration-sample-script located in opendevstack.github.io to the parent folder of `ods-configuration-sample` and `ods-configuration`.
```
Comment (rattermeyer) [inline doc/getting-started.md]: which script are you now referring to? have I overlooked a PR? Coud you provide a URL?

Comment (Tobias-Kohl) [reply doc/getting-started.md]: The Script refers to the Script which will be in ods-core. I will update that to point to ods-core.

Comment (tjaeschke) [reply doc/getting-started.md]: It now seems to be ready for approval, isn't it

Comment (Tobias-Kohl) [reply doc/getting-started.md]: Yes, should be ready for approval.

File: doc/getting-started.md [hunk 2]
```diff
@@ -595,6 +595,8 @@ find ods-configuration -name '*.sample' -type f | while read NAME ; do mv "${NAM
 
 Now you will have to check the `.env` configuration files in `ods-configuration`. Change all values with the suffix `_base64` to a Base64 encoded value.  
 
+If you want later changes to the repositories, update `ods-configuration-sample` and copy the configuration-sample.sh script located in ods-core/infrastructure-setup/scripts to the parent folder of `ods-configuration-sample` and `ods-configuration`.
```
Comment (michaelsauter) [inline doc/getting-started.md:598]: Hmm. Do we really need to copy? Why not keep it in `ods-core`? And if we want to have a different location, then I'd suggest to symlink ...

**Review by michaelsauter — CHANGES_REQUESTED** [review]

**Comment by tjaeschke** [comment]
Closed, because it is reworked completely

**Pull request closed without merging**